speechbrain/lobes/models/huggingface_transformers/speecht5.py

```diff
@@ -0,0 +1,483 @@
+"""This lobe enables the integration of huggingface pretrained SpeechT5 models for Automatic Speech Recognition.
+Reference: https://aclanthology.org/2022.acl-long.393.pdf
+
+Transformer from HuggingFace needs to be installed:
+https://huggingface.co/transformers/installation.html
+
+Authors
+ * Haroun Elleuch 2024
+"""
+
+import gc
+import logging
+from typing import Tuple
+
+import torch
+from transformers import (
+    GenerationConfig,
+    SpeechT5Config,
+    SpeechT5ForSpeechToText,
+)
+
+from speechbrain.lobes.models.huggingface_transformers.huggingface import (
+    HFTransformersInterface,
+)
+from speechbrain.utils.checkpoints import (
+    mark_as_loader,
+    mark_as_saver,
+    register_checkpoint_hooks,
+)
+from speechbrain.utils.fetching import fetch
+
+logger = logging.getLogger(__name__)
+
+
+@register_checkpoint_hooks
+class SpeechT5ForASR(HFTransformersInterface):
```

Comment:
BTW, about the name of this class. How different is this class from SpeechT5 for MT etc ? I mean, what makes this class tailored for ASR only?

Reply:
Yes, in fct there is a difference. SpeechT5 is pretrained to perform both STT and TTS tasks. This model is explicitly for STT.

Reply:
Can't we have a base SpeechT5 class and then have an extension for ASR?